vmm/src/device_manager.rs

```diff
@@ -492,7 +492,7 @@ pub fn create_pty(non_blocking: bool) -> io::Result<(File, File, PathBuf)> {
     // See https://www.kernel.org/doc/Documentation/filesystems/devpts.txt
     // for further details.
 
-    let custom_flags = libc::O_NOCTTY | if non_blocking { libc::O_NONBLOCK } else { 0 };
+    let custom_flags = libc::O_NONBLOCK;
```

Comment:
Oh yes I've done that because it doesn't make sense to apply this flag on the ptmx side of the PTY. You can refer to what `glibc` does here: https://elixir.bootlin.com/glibc/glibc-2.36/source/sysdeps/unix/sysv/linux/getpt.c#L38

The `O_NOCTTY` flag makes sense for the other end of the PTY, which we don't want to set as the controlling terminal.